shared/test/TestSymbolAnimationCoordinatorMap.cpp

```diff
@@ -0,0 +1,190 @@
+        auto second = coordinatorMap.getOrAddAnimationController(1, Coord(0, 0, 0, 0), 1, 0.0, 0.0, 10, 0);
+        REQUIRE(first == second);
+    }
+
+    SECTION("clearAnimationCoordinators removes unused coordinators") {
+        auto first = coordinatorMap.getOrAddAnimationController(1, Coord(0, 0, 0, 0), 0, 0.0, 0.0, 10, 0);
+        coordinatorMap.clearAnimationCoordinators();
+        auto second = coordinatorMap.getOrAddAnimationController(1, Coord(0, 0, 0, 0), 1, 0.0, 0.0, 10, 0);
+        REQUIRE(first != second);
+    }
+
+    SECTION("clearAnimationCoordinators removes only unused coordinators") {
+        auto first = coordinatorMap.getOrAddAnimationController(1, Coord(0, 0, 0, 0), 0, 0.0, 0.0, 10, 0);
+        first->increaseUsage();
+        coordinatorMap.clearAnimationCoordinators();
+        auto second = coordinatorMap.getOrAddAnimationController(1, Coord(0, 0, 0, 0), 1, 0.0, 0.0, 10, 0);
+        REQUIRE(first == second);
+    }
+};
+
+TEST_CASE("SymbolAnimationCoordinatorMap with multiple threads") {
+
+    BENCHMARK("Benchmark random data") {
+        std::atomic<size_t> counter{0};
+        int callsNum = 1000;
+        size_t numThreads = std::thread::hardware_concurrency();
+
+        SymbolAnimationCoordinatorMap coordinatorMap;
+
+        auto threadFunc = [&coordinatorMap, &counter, &callsNum](size_t threadIndex) {
+            for (size_t i = 0; i < callsNum; ++i) {
+                size_t crossTileIdentifier = threadIndex * 1000 + i;
+                Coord coord{static_cast<int32_t>(static_cast<double>(threadIndex)), static_cast<double>(i), 0, 0};
+                int zoomIdentifier = static_cast<int>(threadIndex % 10);
+                double xTolerance = 0.1;
+                double yTolerance = 0.1;
+                int64_t animationDuration = 300;
+                int64_t animationDelay = 0;
+
+                coordinatorMap.getOrAddAnimationController(crossTileIdentifier, coord, zoomIdentifier, xTolerance, yTolerance,
+                                                           animationDuration, animationDelay);
+                ++counter;
+            }
+        };
+
+        std::vector<std::thread> threads;
+
+        for (size_t i = 0; i < numThreads; ++i) {
+            threads.emplace_back(threadFunc, i);
+        }
+
+        for (auto &thread : threads) {
+            thread.join();
+        }
+
+        REQUIRE(counter == numThreads * callsNum);
+    };
+
+
+    auto logEntries = readLogFile("data/symbolAnimationCoordinator/recording.txt");
+
+    BENCHMARK("Simulate from log file") {
+        std::atomic<size_t> counter{0};
+        size_t numThreads = std::thread::hardware_concurrency();
+
+        int runCount = 1;
+
+        SymbolAnimationCoordinatorMap coordinatorMap;
+
+        auto threadFunc = [&coordinatorMap, &counter, &runCount, &logEntries](size_t threadIndex, size_t numThreads) {
+            std::random_device rd;
+            std::mt19937 g(rd());
```

Comment:
Initialize this with `Catch::getSeed()` to get reproducible randomness -- the seed is printed in the log and can be fed back into the test runner with `--rng-seed` to reproduce a particular run.